doc/platform_api/CMIS_and_C-CMIS_support_for_ZR.md

```diff
@@ -0,0 +1,762 @@
+          135: ['CD low granularity, long link [ps/nm]', 'S16', 20],
+          136: ['DGD [ps]', 'U16', 0.01],
+          137: ['SOPMD [ps^2]', 'U16', 0.01],
+          138: ['PDL [dB]', 'U16', 0.1],
+          139: ['OSNR [dB]', 'U16', 0.1],
+          140: ['eSNR [dB]', 'U16', 0.1],
+          141: ['CFO [MHz]', 'S16', 1],
+          142: ['EVM_modem [%]', 'U16', 100.0/65535],
+          143: ['Tx Power [dBm]', 'S16', 0.01],
+          144: ['Rx Total Power [dBm]', 'S16', 0.01],
+          145: ['Rx Signal Power [dBm]', 'S16', 0.01],
+          146: ['SOP ROC [krad/s]', 'U16', 1],
+          147: ['MER [dB]', 'U16', 0.1]
+}
+```
+-  C-CMIS related pages (Page 30h - 4Fh, C-CMIS)
+
+See [C-CMIS v1.1](https://www.oiforum.com/wp-content/uploads/OIF-C-CMIS-01.1.pdf)
+
+|Address|Page Description|Type|
+|-------|----------------|----|
+|30h|Media Lane Configurable Thresholds|RW|
+|31h|Media Lane Provisioning|RW|
+|32h|Media Lane Masks|RW|
+|33h|Media Lane Flags|RO/COR|
+|34h|Media Lane FEC PM|RO|
+|35h|Media Lane Link PM|RO|
+|38h|Host Interface Configuration|RW|
+|3Ah|Host Interface PM|RO|
+|3Bh|Host Interface Flags and Masks|mixed|
+|41h|RX Power Advertisement and Configurable Thresholds|RO|
+|42h|PM Advertisement|RO|
+|43h|Media Lane Provisioning Advertisement|RO|
+
+### Method to read from and write to registers
+
+#### Read and write registers
+- read_reg
+- write_reg
+
+Read and write registers use vendor provided functions to access the bottom layer registers with dictionaries defined Sample code to read and write registers with vendor provided functoins. As mentioned above, the address of a byte with *page* and *offset* is *page* * 128 + *offset*. Below is sample code to read and write registers.
+
+```
+import sonic_platform.platform
+import sonic_platform_base.sonic_sfp.sfputilhelper
+platform_chassis = sonic_platform.platform.Platform().get_chassis()
+PAGE_SIZE = 128
+
+def read_reg(port, page, offset, size):
+    return platform_chassis.get_sfp(port).read_eeprom(page*PAGE_SIZE + offset,size)
+
+def write_reg(port, page, offset, size, write_raw):
+    platform_chassis.get_sfp(port).write_eeprom(page*PAGE_SIZE + offset,size,write_raw)
+```
+#### Encoding and decoding raw data
+- read_reg_from_dict
+- write_reg_from_dict
+
+Read and write registers from dictionary use dictionaries defined [here](#definition-on-cmis-and-c-cmis-registers) and calls read and write register function [here](#read-and-write-registers). We use Python built-in library ```struct``` to encode and decode the raw data in bytearray form to meaningful values. Sample code to decode and encode from/to rawdata in registers:
+
+```
+def read_reg_from_dict(port, Dict): 
+    read_raw = read_reg(port, page = Dict['PAGE'], offset = Dict['OFFSET'], size = Dict['SIZE'])
+    read_buffer = struct.unpack(Dict['TYPE'], read_raw)
+    if len(read_buffer) == 1:
+        return read_buffer[0]
+    else:
+        return read_buffer
+
+def write_reg_from_dict(port, Dict, write_buffer):
```

Comment:
How does different vendors support/extend these interfaces/api?

Reply:
updated the HLD. we can use xcvrd but the bottom layer still access module registers from APIs in sonic-platform-common repo.